src/main/java/org/apache/sling/resourceresolver/impl/mapping/MapEntries.java

```diff
@@ -1288,50 +1288,60 @@ private Iterator<Resource> queryAllVanityPaths(String query) {
         return it;
     }
 
+    /**
+     * Utility class for running paged queries.
+     */
     private class PagedQueryIterator implements Iterator<Resource> {
 
         private ResourceResolver resolver;
+        private String name;
+        private String propertyName;
         private String query;
-        private String lastPath = "";
+        private String lastValue = "";
         private Iterator<Resource> it;
         private int count = 0;
         private int page = 0;
-        private int pageSize = Integer.getInteger("sling.vanityPath.pageSize", 2000);
+        private int pageSize;
         private Resource next = null;
+        private String[] defaultValue = new String[0];
 
-        public PagedQueryIterator(ResourceResolver resolver, String query) {
+        public PagedQueryIterator(String name, String propertyName, ResourceResolver resolver, String query, int pageSize) {
```

Comment:
```name``` is quite generic. If we don't come up with a better name :-) we should at least document its purpose here.